src/CreatePhotosSlideshow/CreatePhotosSlideshow.tsx

```diff
@@ -130,10 +130,11 @@ const CreatePhotosSlideshow = (): JSX.Element => {
   }, [correctedFileArrayBuffer, files, isGenerating, orderedItems]);
 
   const onDrop = useCallback(
-    async (files: File[]) => {
-      setFiles(files);
-      setItems(new Array(files.length).fill(0).map((_, index) => index));
-      for (const file of files) {
+    async (newFiles: File[]) => {
+      const newFilesLength = newFiles.length + files.length;
+      setFiles((oldFiles) => [...oldFiles, ...newFiles]);
+      setItems(new Array(newFilesLength).fill(0).map((_, index) => index));
```

Comment:
This will override any re-ordering the user has already done before adding new files. We addressed this in the `CombilePDF.tsx` tool if you wanna take a look. But I am going to merge this for now, feel free to send a follow up PR to fix this.